forem/forem
# Improve the UX of the SMTP Section

This branch is not names aptly as it metamorphosed into a different feature.

## What type of PR is this? (check all applicable)


## Description

In this PR:

I rename the SMTP Header on the config page and make some copy changes. I also make a significant UX change that determines whether to display a 'use your own server' toggle, some description text and the SMTP Form based on whether the user has a Sendgrid API Key and/or custom SMTP Settings. 

This is how the feature should behave: 

| Sendgrid API Key | Custom SMTP Settings | Show Toggle Element  | Text description                                                                                                                                                                                                                                  | Action                                                                               |
|------------------|----------------------|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------|
| no               | no                   | no                   | none                                                                                                                                                                                                                                              | Show the SMTP Form                                                                   |
| yes              | no                   | yes                  | As a Forem Cloud client, we provide an email server managed by the Forem team. All settings are managed by us and the from and reply email addresses are set as noreply@forem.com. . However, you can override this to use your own email server. | When the user toggles the box, the SMTP Form will show up                            |
| no               | yes                  | no                   | none                                                                                                                                                                                                                                              | Show the SMTP Form                                                                   |
| yes              | yes                  | yes                  | none                                                                                                                                                                                                                                              | A user can un toggle and it should clear out the form and revert to the sengrid key. |


You can read more about the discussion [here](https://github.com/forem/forem/issues/16180).

In a subsequent PR I'll add a reply_to address, a from_email address and a contact email address. 

You can find a demo video here:
https://www.loom.com/share/472c958d890140dfa13400ee3f054a10

## Related Tickets & Documents
https://github.com/forem/forem/issues/16180

## QA Instructions, Screenshots, Recordings

You can find a demo video here:
https://www.loom.com/share/472c958d890140dfa13400ee3f054a10

In the table above where it says `SENDGRID_API_KEY` needs to be set. You will need to set your  `SENDGRID_API_KEY` in env.yml. When you add or remove your ENV key you will need to restart the server.

For the  'yes'.  values in `Custom SMTP Settings` column, you will need to set the minimum fields required for SMTP: address, username, password.  

Navigate to /admin/customization/config` and ensure that what you see in the Email Server Section is based on the table in the description.
 
### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
